django_app/redbox_app/redbox_core/models.py

```diff
@@ -64,10 +66,20 @@ class File(UUIDPrimaryKeyBase, TimeStampedModel):
     user = models.ForeignKey(User, on_delete=models.CASCADE)
     original_file_name = models.TextField(max_length=2048, blank=True, null=True)
     core_file_uuid = models.UUIDField(null=True)
+    expiry_date = models.DateTimeField(blank=True, null=True)
 
     def __str__(self) -> str:  # pragma: no cover
         return f"{self.original_file_name} {self.user}"
 
+    def save(self, *args, **kwargs):
+        if not self.expiry_date:
+            if self.created_at:
+                #  Needed to populate the initial expiry date for existing Files
+                self.expiry_date = self.created_at + timedelta(seconds=settings.FILE_EXPIRY_IN_SECONDS)
+            else:
```

Comment:
i dont think that the `created_at` is ever null?

Reply:
It is on the first save() of creating an object. I tried running `super.save()` before this, but it needs running again to store the expiry_date, which causes pk uniqueness issues. The other option is to use a signal, but as mentioned above, Django doesn't recommend it.

Reply:
TIL. ok, this all makes sense